### ISSUE TITLE
[A11Y] Select Component - Content does not reflow at 320px width equivalent

### ISSUE THREAD

**Issue by DelishusCake** [dequelabs/cauldron#1622]
### Short Description

The `Select` component does not reflow to fit its container at a horizontal viewport size of 320px.

### Impact

- [ ] Blocker
- [ ] Critical
- [x] Serious
- [ ] Moderate
- [ ] Minor

### Steps To Reproduce:

1. Add a `Select` component to a container
2. Set the browser horizontal viewport size to 320px

### Current Behavior:

The `Select` component maintains a minimum width of 300px (`--select-width`)

### Expected Behavior:

The `Select` component should reflow to fit its container at low viewport sizes.

---

## Additional Context

A similar issue (#579) was opened a while ago, however the fix does not seem to have removed the minimum width for the component.

### Proposed solution
[Remove the minimum width for the component wrapper](https://github.com/dequelabs/cauldron/blob/53ae3d05aeb1b6263b33faf44c576ced7c814608/packages/styles/select.css#L19) or insert a media-query to only enforce this minimum-width at larger viewport sizes.

---

## Screenshots 
![image](https://github.com/user-attachments/assets/7acd8423-4520-46b5-aade-f8c900c57330)
![image](https://github.com/user-attachments/assets/e1ca018e-86b7-439d-ad74-86307a4568ee)

---

### 1. Perceivable

<details> 
<summary>Expand</summary>

#### 1.1 Text Alternatives

- [ ] 1.1.1 Non-text Content

#### 1.2 Time-based Media

- [ ] 1.2.1 Audio-only and Video-only (Prerecorded)
- [ ] 1.2.2 Captions (Prerecorded)
- [ ] 1.2.3 Audio Description or Media Alternative (Prerecorded)
- [ ] 1.2.4 Captions (Live)
- [ ] 1.2.5 Audio Description (Prerecorded)

#### 1.3 Info and Relationships

- [ ] 1.3.1 Info and Relationships
- [ ] 1.3.2 Meaningful Sequence
- [ ] 1.3.3 Sensory Characteristics
- [ ] 1.3.4 Orientation
- [ ] 1.3.5 Identify Input Purpose

#### 1.4 Distinguishable

- [ ] 1.4.1 Use of Color
- [ ] 1.4.2 Audio Control
- [ ] 1.4.3 Contrast (Minimum)
- [ ] 1.4.4 Resize text
- [ ] 1.4.5 Images of Text
- [x] 1.4.10 Reflow
- [ ] 1.4.11 Non-text Contrast
- [ ] 1.4.12 Text Spacing
- [ ] 1.4.13 Content on Hover or Focus
</details>

### 2. Operable

<details> 
<summary>Expand</summary>

#### 2.1 Keyboard Accessible

- [ ] 2.1.1 Keyboard
- [ ] 2.1.2 No Keyboard Trap
- [ ] 2.1.4 Character Key Shortcuts

#### 2.2 Enough Time

- [ ] 2.2.1 Timing Adjustable
- [ ] 2.2.2 Pause, Stop, Hide

#### 2.3 Seizures and Physical Reactions

- [ ] 2.3.1 Three Flashes or Below Threshold

#### 2.4 Navigable

- [ ] 2.4.1 Bypass Blocks
- [ ] 2.4.2 Page Titled
- [ ] 2.4.3 Focus Order
- [ ] 2.4.4 Link Purpose (In Context)
- [ ] 2.4.5 Multiple Ways
- [ ] 2.4.6 Headings and Labels
- [ ] 2.4.7 Focus Visible

#### 2.5 Input Modalities

- [ ] 2.5.1 Pointer Gestures
- [ ] 2.5.2 Pointer Cancellation
- [ ] 2.5.3 Label in Name
- [ ] 2.5.4 Motion Actuation
</details>

### 3. Understandable

<details> 
<summary>Expand</summary>

#### 3.1 Readable

- [ ] 3.1.1 Language of Page
- [ ] 3.1.2 Language of Parts

#### 3.2 Predictable

- [ ] 3.2.1 On Focus
- [ ] 3.2.2 On Input
- [ ] 3.2.3 Consistent Navigation
- [ ] 3.2.4 Consistent Identification

#### 3.3 Input Assistance

- [ ] 3.3.1 Error Identification
- [ ] 3.3.2 Labels or Instructions
- [ ] 3.3.3 Error Suggestion
- [ ] 3.3.4 Error Prevention (Legal, Financial, Data)
</details>

### 4. Robust

<details>
<summary>Expand</summary>

#### 4.1 Compatible

- [ ] 4.1.1 Parsing
- [ ] 4.1.2 Name, Role, Value
- [ ] 4.1.3 Status Messages
</details>

**Comment by scurker** [comment]
@DelishusCake Are you sure this is a Cauldron issue? I checked the select component in docs and it looks like it does collapse correctly when it cannot fill the container:

![image](https://github.com/user-attachments/assets/2d118e69-1859-4d53-ba5a-44539c9d3f05)

Is there any additional information that you can provide? My guess is the select component is in a container that doesn't have a constrained width so it's trying to take up the maximum space that it's allowed.

**Comment by DelishusCake** [comment]
Hmm. I think this issue might actually stem from the `Popover` component. Both the `Select` and `Textfield` components reflow fine when contained within a `Panel`, but seem to have issues when contained within a `Popover`.
![image](https://github.com/user-attachments/assets/d4b5f3e8-9ae3-4346-b78f-8cc7900d08cc)

I'm going to put together a minimal replication and see if the issue lies in our use of the `Popover` component, or is an issue with the `Popover` component more generally.

**Comment by DelishusCake** [comment]
I'm going to close this for now, this seems to be two different issues that happen to be manifesting in the same visual way. I'll re-open this ticket or create a new on if my investigation turns up a root issue that happens to be in Cauldron
